File: app/src/main/java/com/commit451/gitlab/viewHolder/IssueViewHolder.java
package com.commit451.gitlab.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.commit451.gitlab.App;
import com.commit451.gitlab.R;
import com.commit451.gitlab.model.api.Issue;
import com.commit451.gitlab.transformation.CircleTransformation;
import com.commit451.gitlab.util.DateUtil;
import com.commit451.gitlab.util.ImageUtil;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * issues, yay!
 */
public class IssueViewHolder extends RecyclerView.ViewHolder {

    public static IssueViewHolder inflate(ViewGroup parent) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_issue, parent, false);
        return new IssueViewHolder(view);
    }

    @BindView(R.id.issue_state)
    TextView textState;
    @BindView(R.id.issue_image)
    ImageView image;
    @BindView(R.id.issue_message)
    TextView textMessage;
    @BindView(R.id.issue_creator)
    TextView textCreator;

    public IssueViewHolder(View view) {
        super(view);
        ButterKnife.bind(this, view);
    }

    public void bind(Issue issue) {

        switch (issue.getState()) {
            case Issue.STATE_OPENED:
                textState.setText(itemView.getResources().getString(R.string.issue_open));
                break;
            case Issue.STATE_CLOSED:
                textState.setText(itemView.getResources().getString(R.string.issue_closed));
                break;
            default:
                textState.setVisibility(View.GONE);
                break;
        }

        if (issue.getAssignee() != null) {
            App.get().getPicasso()
                    .load(ImageUtil.getAvatarUrl(issue.getAssignee(), itemView.getResources().getDimensionPixelSize(R.dimen.image_size)))
                    .transform(new CircleTransformation())
                    .into(image);
        } else {
            image.setImageBitmap(null);
        }

        textMessage.setText(issue.getTitle());

        String time = "";
        if (issue.getCreatedAt() != null) {
            time += DateUtil.getRelativeTimeSpanString(itemView.getContext(), issue.getCreatedAt());
        }
        String author = "";
        if (issue.getAuthor() != null) {
            author += issue.getAuthor().getUsername();
        }
        String id = "";
        long issueId = issue.getIid();
        if (issueId < 1) {
            issueId = issue.getId();
        }
        id = "#" + issueId;

        textCreator.setText(String.format(itemView.getContext().getString(R.string.opened_time), id, time, author));
    }
}

File: app/src/main/java/com/commit451/gitlab/viewHolder/IssueViewHolder.kt
package com.commit451.gitlab.viewHolder

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.commit451.gitlab.App
import com.commit451.gitlab.R
import com.commit451.gitlab.model.api.Issue
import com.commit451.gitlab.transformation.CircleTransformation
import com.commit451.gitlab.util.DateUtil
import com.commit451.gitlab.util.ImageUtil

/**
 * issues, yay!
 */
class IssueViewHolder(view: View) : RecyclerView.ViewHolder(view) {

    companion object {

        fun inflate(parent: ViewGroup): IssueViewHolder {
            val view = LayoutInflater.from(parent.context)
                    .inflate(R.layout.item_issue, parent, false)
            return IssueViewHolder(view)
        }
    }

    @BindView(R.id.issue_state)
    lateinit var textState: TextView
    @BindView(R.id.issue_image)
    lateinit var image: ImageView
    @BindView(R.id.issue_message)
    lateinit var textMessage: TextView
    @BindView(R.id.issue_creator)
    lateinit var textCreator: TextView

    init {
        ButterKnife.bind(this, view)
    }

    fun bind(issue: Issue) {

        when (issue.state) {
            Issue.STATE_OPENED -> textState.text = itemView.resources.getString(R.string.issue_open)
            Issue.STATE_CLOSED -> textState.text = itemView.resources.getString(R.string.issue_closed)
            else -> textState.visibility = View.GONE
        }

        if (issue.assignee != null) {
            App.get().picasso
                    .load(ImageUtil.getAvatarUrl(issue.assignee, itemView.resources.getDimensionPixelSize(R.dimen.image_size)))
                    .transform(CircleTransformation())
                    .into(image)
        } else {
            image.setImageBitmap(null)
        }

        textMessage.text = issue.title
        textMessage.setCompoundDrawablesWithIntrinsicBounds(0, 0, 0, 0)
        if (issue.isConfidential) {
            textMessage.setCompoundDrawablesWithIntrinsicBounds(R.drawable.ic_visibility_off_white_24dp, 0, 0, 0)
        }

        var time = ""
        if (issue.createdAt != null) {
            time += DateUtil.getRelativeTimeSpanString(itemView.context, issue.createdAt)
        }
        var author = ""
        if (issue.author != null) {
            author += issue.author!!.username
        }
        val id: String
        var issueId = issue.iid
        if (issueId < 1) {
            issueId = issue.id
        }
        id = "#" + issueId

        textCreator.text = String.format(itemView.context.getString(R.string.opened_time), id, time, author)
    }
}
